proxy/election.go

```diff
@@ -453,6 +454,61 @@ func (h *election) Elections(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// DeleteElection implements proxy.Proxy
+func (h *election) DeleteElection(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	if vars == nil || vars["electionID"] == "" {
+		http.Error(w, fmt.Sprintf("electionID not found: %v", vars), http.StatusInternalServerError)
+		return
+	}
+
+	electionID := vars["electionID"]
+
+	elecMD, err := h.getElectionsMetadata()
+	if err != nil {
+		http.Error(w, "failed to get election metadata", http.StatusNotFound)
+		return
+	}
+
+	if elecMD.ElectionsIDs.Contains(electionID) < 0 {
+		http.Error(w, "the election does not exist", http.StatusNotFound)
+		return
+	}
+
+	// auth should contain the hex-encoded signature on the hex-encoded election
+	// ID
+	auth := r.Header.Get("Authorization")
+
+	sig, err := hex.DecodeString(auth)
+	if err != nil {
+		BadRequestError(w, r, xerrors.Errorf("failed to decode auth: %v", err), nil)
+		return
+	}
+
+	err = schnorr.Verify(suite, h.pk, []byte(electionID), sig)
+	if err != nil {
+		ForbiddenError(w, r, xerrors.Errorf("signature verification failed: %v", err), nil)
+		return
+	}
+
+	deleteElection := types.DeleteElection{
+		ElectionID: electionID,
+	}
+
+	data, err := deleteElection.Serialize(h.context)
+	if err != nil {
+		InternalError(w, r, xerrors.Errorf("failed to marshal DeleteElection: %v", err), nil)
+		return
+	}
+
+	_, err = h.submitAndWaitForTxn(r.Context(), evoting.CmdDelteElection, evoting.ElectionArg, data)
+	if err != nil {
+		http.Error(w, "failed to submit txn: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
```

Comment:
I'm not entirely familiar with the new format of the proxy yet, we don't need to write back here?

Reply:
No, there is no data to return. Of course if something went wrong the proxy will return an HTTP error (error 500).